Stresser/ArtifactManager.cpp

```diff
@@ -1,17 +1,45 @@
 #include "ArtifactManager.h"
 #include "ArtifactFactory.h"
 
-ArtifactManager::ArtifactManager()
-{
-}
-
 void ArtifactManager::policyChanged(const std::vector<RuleEntity>& rules)
 {
+	// Uninstall the old artifacts
+	for (auto subscriber : this->m_subscribers)
+	{
+		subscriber->reset();
+	}
 	this->m_artifactsVector.clear();
-	
+
+	// Getting the new artifacts
 	for (const RuleEntity& ruleEntity : rules)
 	{
 		auto artifact = ArtifactFactory::BuildArtifact(ruleEntity.getType(), ruleEntity.getName(), ruleEntity.getData());
 		this->m_artifactsVector.push_back(std::move(artifact));
 	}
+
+	// Update the subscribers artifacts
+	for (const auto subscriber : this->m_subscribers)
+	{
+		subscriber->setNewArtifacts(this->getArtifactsByType(subscriber->getType()));
+	}
+}
+
+std::vector<IArtifact*> ArtifactManager::getArtifactsByType(const std::string& type)
+{
+	std::vector<IArtifact*> result;
+
+	for (const auto& artifact :  this->m_artifactsVector)
+	{
+		if (artifact->getType() == type)
```

Comment:
I've wrote a few comments about the `type` member.
We should remove it if possible.

One way of doing it is trying to convert the type to the derived class: 
https://stackoverflow.com/a/500495

More generic way:
https://www.tutorialspoint.com/cplusplus-equivalent-of-instanceof